### ISSUE TITLE
Build fail due to apt update failure (GPG error: https://debian.neo4j.org/repo stable/ Release: The following signatures were invalid: BADSIG 26C95CF201182252 Neo Technology Admins <admins@neotechnology.com>)

### ISSUE THREAD

**Issue by Lin-Buo-Ren** [travis-ci/travis-ci#7987]
[Build #7 - Lin-Buo-Ren/Clean-Filter-for-XML - Travis CI](https://travis-ci.org/Lin-Buo-Ren/Clean-Filter-for-XML/builds/247878321)

It happens at least since 2017-06-28T10:17:09Z, while 2017-06-28T09:47:25Z still working

UPDATE: [Travis CI Status - Apt failures](https://www.traviscistatus.com/incidents/fyskznm7wg2c)
UPDATE: According to Travis CI Status a hotfix(?) is in place for this issue
UPDATE: The issue is no longer reproduced after the hotfix

**Comment by Lin-Buo-Ren** [comment]
I have contacted them via the [contact-us form](https://neo4j.com/contact-us/).

**Comment by leezer3** [comment]
+1
https://travis-ci.org/leezer3/OpenBVE/jobs/247889512

**Comment by Lin-Buo-Ren** [comment]
I wonder if it's due to the key switch to deal with the deprecated SHA-1 issue

[Neo4j Debian Packages](http://debian.neo4j.org/)

**Comment by tbarbugli** [comment]
is there any workaround for this?

**Comment by Lin-Buo-Ren** [comment]
@tbarbugli I suspect that avoid installing anything using the APT addon will work

**Comment by Lin-Buo-Ren** [comment]
Since the issue is no longer reproduced, I'll close it for now.
Thanks for the prompt fix for Travis CI.

**Comment by nikolas** [comment]
I'm getting repeated neo4j.org hash sum mismatches on this build: https://travis-ci.org/ccnmtl/blackrock/builds/248530281

```
W: http://dl.hhvm.com/ubuntu/dists/trusty/InRelease: Signature by key 36AEF64D0207E7EEE352D4875A16E7281BE7A449 uses weak digest algorithm (SHA1)
W: http://ppa.launchpad.net/couchdb/stable/ubuntu/dists/trusty/Release.gpg: Signature by key 15866BAFD9BCC4F3C1E0DFC7D69548E1C17EAB57 uses weak digest algorithm (SHA1)
E: Failed to fetch https://debian.neo4j.org/repo/stable/Packages.gz  Hash Sum mismatch
E: Some index files failed to download. They have been ignored, or old ones used instead.
```

**Comment by BanzaiMan** [comment]
The issue appears to have resurfaced.

**Comment by Geolim4** [comment]
@BanzaiMan I confirm that the issue resurfaced:
HHVM build are broken: https://travis-ci.org/PHPSocialNetwork/phpfastcache/jobs/248553863

:)

**Comment by BanzaiMan** [comment]
See also: http://stspg.io/1d8f4b1fd

**Comment by edmorley** [comment]
It looks like travis-ci/travis-build#1090 was manually deployed to production but not merged to master, so may have then been overwritten by an unrelated PR being merged to master 3 hours ago (presuming master auto-deploys).

**Comment by BanzaiMan** [comment]
@edmorley Thanks for pointing that out. A deployment earlier today overwrote that hot fix. I fixed the PR, merged and deployed it to production. The fix should be live now. Please test your builds. Restarting should be sufficient.

The neo4j source is unavailable as a result, and we will look at the remedy for this, too.

**Comment by BanzaiMan** [comment]
I'm closing this again. The PR to remove neo4j apt source has been merged and deployed. The neo4j source needs to be reinstated, but that will be tracked separately in #8002.

**Comment by JCorson** [comment]
I am currently running into an error similar to #7998 (which was closed as a duplicate of this issue). Any ideas?

```
W: Target Packages (stable/binary-amd64/Packages) is configured multiple times in /etc/apt/sources.list:25 and /etc/apt/sources.list.d/docker.list:1
W: Target Packages (stable/binary-all/Packages) is configured multiple times in /etc/apt/sources.list:25 and /etc/apt/sources.list.d/docker.list:1
W: Target Translations (stable/i18n/Translation-en_US) is configured multiple times in /etc/apt/sources.list:25 and /etc/apt/sources.list.d/docker.list:1
W: Target Translations (stable/i18n/Translation-en) is configured multiple times in /etc/apt/sources.list:25 and /etc/apt/sources.list.d/docker.list:1
W: http://dl.hhvm.com/ubuntu/dists/trusty/InRelease: Signature by key 36AEF64D0207E7EEE352D4875A16E7281BE7A449 uses weak digest algorithm (SHA1)
N: Skipping acquire of configured file 'stable/binary-i386/Packages' as repository 'https://download.docker.com/linux/ubuntu trusty InRelease' doesn't support architecture 'i386'
N: Skipping acquire of configured file 'edge/binary-i386/Packages' as repository 'https://download.docker.com/linux/ubuntu trusty InRelease' doesn't support architecture 'i386'
W: http://ppa.launchpad.net/couchdb/stable/ubuntu/dists/trusty/Release.gpg: Signature by key 15866BAFD9BCC4F3C1E0DFC7D69548E1C17EAB57 uses weak digest algorithm (SHA1)
W: http://ppa.launchpad.net/builds/sphinxsearch-rel20/ubuntu/dists/trusty/Release.gpg: Signature by key 2FE7BA61B28B34B7A2B749A2B9D8946B16932B16 uses weak digest algorithm (SHA1)
W: http://ppa.launchpad.net/builds/sphinxsearch-rel21/ubuntu/dists/trusty/Release.gpg: Signature by key 2FE7BA61B28B34B7A2B749A2B9D8946B16932B16 uses weak digest algorithm (SHA1)
W: Target Packages (stable/binary-amd64/Packages) is configured multiple times in /etc/apt/sources.list:25 and /etc/apt/sources.list.d/docker.list:1
W: Target Packages (stable/binary-all/Packages) is configured multiple times in /etc/apt/sources.list:25 and /etc/apt/sources.list.d/docker.list:1
W: Target Translations (stable/i18n/Translation-en_US) is configured multiple times in /etc/apt/sources.list:25 and /etc/apt/sources.list.d/docker.list:1
W: Target Translations (stable/i18n/Translation-en) is configured multiple times in /etc/apt/sources.list:25 and /etc/apt/sources.list.d/docker.list:1
```

**Comment by BanzaiMan** [comment]
Those are warnings, not errors. If your builds are failing, they're falling for a different reason.

**Comment by JCorson** [comment]
Ah yes, thanks for the correction @BanzaiMan. However, the failure that we are seeing does seem to be related to these:

```
N: Skipping acquire of configured file 'stable/binary-i386/Packages' as repository 'https://download.docker.com/linux/ubuntu trusty InRelease' doesn't support architecture 'i386'
N: Skipping acquire of configured file 'edge/binary-i386/Packages' as repository 'https://download.docker.com/linux/ubuntu trusty InRelease' doesn't support architecture 'i386'
```

The failure is in an ansible playbook and is failing with the following traceback:

```
failed: [travis-ci] => (item=deb http://ppa.launchpad.net/adiscon/v8-stable/ubuntu trusty main) => {"failed": true, "item": "deb http://ppa.launchpad.net/adiscon/v8-stable/ubuntu trusty main", "parsed": false}
...
  File "/usr/lib/python2.7/dist-packages/apt/cache.py", line 440, in update
    raise FetchFailedException(e)
apt.cache.FetchFailedException: W:Failed to fetch https://download.docker.com/linux/ubuntu/dists/trusty/InRelease  Unable to find expected entry 'stable/binary-i386/Packages' in Release file (Wrong sources.list entry or malformed file)
, E:Some index files failed to download. They have been ignored, or old ones used instead.
```

The build passes when `group: deprecated-2017Q2` is specified.

**Comment by Blackbaud-BrandonJones** [comment]
Resurfaced?
```
Reading package lists... Done
W: http://dl.hhvm.com/ubuntu/dists/trusty/InRelease: Signature by key 36AEF64D0207E7EEE352D4875A16E7281BE7A449 uses weak digest algorithm (SHA1)
W: http://ppa.launchpad.net/couchdb/stable/ubuntu/dists/trusty/Release.gpg: Signature by key 15866BAFD9BCC4F3C1E0DFC7D69548E1C17EAB57 uses weak digest algorithm (SHA1)
W: The repository 'https://packagecloud.io/basho/riak/ubuntu trusty Release' does not have a Release file.
N: Data from such a repository can't be authenticated and is therefore potentially dangerous to use.
N: See apt-secure(8) manpage for repository creation and user configuration details.
E: Failed to fetch https://packagecloud.io/basho/riak/ubuntu/dists/trusty/main/source/Sources  401  Unauthorized
E: Failed to fetch https://packagecloud.io/basho/riak/ubuntu/dists/trusty/main/binary-amd64/Packages  401  Unauthorized
E: Failed to fetch https://packagecloud.io/basho/riak/ubuntu/dists/trusty/main/binary-i386/Packages  401  Unauthorized
E: Some index files failed to download. They have been ignored, or old ones used instead.
The command "sudo apt-get update" failed and exited with 100 during .
```

**Comment by leezer3** [comment]
As far as I can tell, the entire RIAK repos have gone MIA.

Exactly what RIAK is doing I'm not sure though.
https://github.com/travis-ci/travis-ci/issues/8604

**Comment by saggu** [comment]
I have started this today,

```
Reading package lists... Done
W: http://dl.hhvm.com/ubuntu/dists/trusty/InRelease: Signature by key 36AEF64D0207E7EEE352D4875A16E7281BE7A449 uses weak digest algorithm (SHA1)
W: http://ppa.launchpad.net/couchdb/stable/ubuntu/dists/trusty/Release.gpg: Signature by key 15866BAFD9BCC4F3C1E0DFC7D69548E1C17EAB57 uses weak digest algorithm (SHA1)
E: Failed to fetch http://dl.hhvm.com/ubuntu/dists/trusty/main/binary-i386/Packages  404  Not Found [IP: 13.33.164.32 80]
E: Some index files failed to download. They have been ignored, or old ones used instead.
The command "sudo apt-get update" failed and exited with 100 during .
Your build has been stopped.z
```

**Comment by michaeltlombardi** [comment]
Confirmed, this problem has popped up in one of ours, too.

```
$ sudo apt-get update
Ign:1 http://us-central1.gce.archive.ubuntu.com/ubuntu trusty InRelease
Hit:2 http://us-central1.gce.archive.ubuntu.com/ubuntu trusty-updates InRelease
Hit:3 http://us-central1.gce.archive.ubuntu.com/ubuntu trusty-backports InRelease
Hit:4 http://us-central1.gce.archive.ubuntu.com/ubuntu trusty Release
Get:5 http://dl.hhvm.com/ubuntu trusty InRelease [2,104 B]
Ign:6 http://repo.mongodb.org/apt/ubuntu trusty/mongodb-org/3.2 InRelease
Ign:7 http://dl.google.com/linux/chrome/deb stable InRelease
Hit:8 http://repo.mongodb.org/apt/ubuntu trusty/mongodb-org/3.2 Release
Hit:9 http://apt.postgresql.org/pub/repos/apt trusty-pgdg InRelease
Hit:10 http://security.ubuntu.com/ubuntu trusty-security InRelease
Ign:11 http://toolbelt.heroku.com/ubuntu ./ InRelease
Get:12 http://dl.bintray.com/apache/cassandra 39x InRelease [3,168 B]
Hit:14 http://ppa.launchpad.net/chris-lea/redis-server/ubuntu trusty InRelease
Hit:13 http://toolbelt.heroku.com/ubuntu ./ Release
Hit:16 http://dl.google.com/linux/chrome/deb stable Release
Get:17 https://packages.microsoft.com/ubuntu/14.04/prod trusty InRelease [2,846 B]
Get:18 http://dl.hhvm.com/ubuntu trusty/main amd64 Packages [1,776 B]
Hit:19 https://download.docker.com/linux/ubuntu trusty InRelease
Ign:20 http://dl.hhvm.com/ubuntu trusty/main i386 Packages
Err:20 http://dl.hhvm.com/ubuntu trusty/main i386 Packages
  404  Not Found [IP: 52.84.14.76 80]
Hit:22 http://ppa.launchpad.net/george-edison55/cmake-3.x/ubuntu trusty InRelease
Ign:23 http://ppa.launchpad.net/couchdb/stable/ubuntu trusty InRelease
Hit:26 http://ppa.launchpad.net/git-core/ppa/ubuntu trusty InRelease
Get:27 https://packages.microsoft.com/ubuntu/14.04/prod trusty/main amd64 Packages [17.4 kB]
Hit:28 https://packagecloud.io/computology/apt-backport/ubuntu trusty InRelease
Hit:29 https://packagecloud.io/github/git-lfs/ubuntu trusty InRelease
Hit:30 http://ppa.launchpad.net/openjdk-r/ppa/ubuntu trusty InRelease
Hit:31 https://packagecloud.io/rabbitmq/rabbitmq-server/ubuntu trusty InRelease
Hit:32 http://ppa.launchpad.net/pollinate/ppa/ubuntu trusty InRelease
Hit:33 http://ppa.launchpad.net/webupd8team/java/ubuntu trusty InRelease
Hit:34 http://ppa.launchpad.net/couchdb/stable/ubuntu trusty Release
Fetched 27.3 kB in 1s (16.5 kB/s)
Reading package lists... Done
W: http://dl.hhvm.com/ubuntu/dists/trusty/InRelease: Signature by key 36AEF64D0207E7EEE352D4875A16E7281BE7A449 uses weak digest algorithm (SHA1)
W: http://ppa.launchpad.net/couchdb/stable/ubuntu/dists/trusty/Release.gpg: Signature by key 15866BAFD9BCC4F3C1E0DFC7D69548E1C17EAB57 uses weak digest algorithm (SHA1)
E: Failed to fetch http://dl.hhvm.com/ubuntu/dists/trusty/main/binary-i386/Packages  404  Not Found [IP: 52.84.14.76 80]
E: Some index files failed to download. They have been ignored, or old ones used instead.
The command "sudo apt-get update" failed and exited with 100 during .
Your build has been stopped.
```

**Comment by goetzc** [comment]
For those arriving here, the issue is already being fixed:
https://www.traviscistatus.com/incidents/rg2rb5mm71vl

**Comment by AndrewAndreev** [comment]
Is it the same issue?
`The command "sudo apt-get -y install qt59-meta-minimal" failed and exited with 100 during .`

**Comment by mikesparr** [comment]
Issue resurfaced again for me today (Jan 18, 2018). Confirm Docker builds work locally. 

 * added `group: deprecated-2017Q2` and does not fix?

## Failed RUN
```
RUN apt-get update && apt-get install -y apt-transport-https && \
    apt-get install -y -qq --force-yes cron && \
    apt-get install -y -qq --force-yes tzdata && \
    apt-get install -y gcsfuse && \
    apt-get install -y libxml2-dev && \
    apt-get install -y php-soap && \
    apt-get install -y libjpeg-dev && \
    apt-get install -y libpng-dev && \
    apt-get install -y libfreetype6-dev && \
    apt-get clean -y && \
    docker-php-ext-install soap && \
    docker-php-ext-install mysqli && \
    docker-php-ext-configure gd --with-freetype-dir=/usr/include --with-jpeg-dir=/user/include --with-png-dir=/usr/include && \
    docker-php-ext-install gd
```

## Errors in build
```
E: Package 'php-soap' has no installation candidate
The command '/bin/sh -c apt-get update && apt-get install -y apt-transport-https &&     apt-get install -y -qq --force-yes cron &&     apt-get install -y -qq --force-yes tzdata &&     apt-get install -y gcsfuse &&     apt-get install -y libxml2-dev &&     apt-get install -y php-soap &&     apt-get install -y libjpeg-dev &&     apt-get install -y libpng-dev &&     apt-get install -y libfreetype6-dev &&     apt-get clean -y &&     docker-php-ext-install soap &&     docker-php-ext-install mysqli &&     docker-php-ext-configure gd --with-freetype-dir=/usr/include --with-jpeg-dir=/user/include --with-png-dir=/usr/include &&     docker-php-ext-install gd' returned a non-zero code: 100
```

## Warnings earlier in build
```
W: Failed to fetch http://ppa.launchpad.net/git-core/ppa/ubuntu/dists/trusty/InRelease  
W: Failed to fetch http://ppa.launchpad.net/openjdk-r/ppa/ubuntu/dists/trusty/InRelease  
W: Failed to fetch http://ppa.launchpad.net/pollinate/ppa/ubuntu/dists/trusty/InRelease  
W: Failed to fetch http://ppa.launchpad.net/rwky/ppa/ubuntu/dists/trusty/InRelease  
W: Failed to fetch http://ppa.launchpad.net/builds/sphinxsearch-rel20/ubuntu/dists/trusty/InRelease  
W: Failed to fetch http://ppa.launchpad.net/builds/sphinxsearch-rel21/ubuntu/dists/trusty/InRelease  
W: Failed to fetch http://ppa.launchpad.net/builds/sphinxsearch-rel22/ubuntu/dists/trusty/InRelease  
W: Failed to fetch http://ppa.launchpad.net/webupd8team/java/ubuntu/dists/trusty/InRelease  
W: Failed to fetch http://ppa.launchpad.net/george-edison55/cmake-3.x/ubuntu/dists/trusty/main/binary-amd64/Packages  Unable to connect to ppa.launchpad.net:http:
W: Failed to fetch http://ppa.launchpad.net/george-edison55/cmake-3.x/ubuntu/dists/trusty/main/i18n/Translation-en  Unable to connect to ppa.launchpad.net:http:
W: Failed to fetch http://ppa.launchpad.net/git-core/ppa/ubuntu/dists/trusty/Release.gpg  Unable to connect to ppa.launchpad.net:http:
W: Failed to fetch http://ppa.launchpad.net/openjdk-r/ppa/ubuntu/dists/trusty/Release.gpg  Unable to connect to ppa.launchpad.net:http:
W: Failed to fetch http://ppa.launchpad.net/pollinate/ppa/ubuntu/dists/trusty/Release.gpg  Unable to connect to ppa.launchpad.net:http:
W: Failed to fetch http://ppa.launchpad.net/rwky/ppa/ubuntu/dists/trusty/Release.gpg  Unable to connect to ppa.launchpad.net:http:
W: Failed to fetch http://ppa.launchpad.net/builds/sphinxsearch-rel20/ubuntu/dists/trusty/Release.gpg  Unable to connect to ppa.launchpad.net:http:
W: Failed to fetch http://ppa.launchpad.net/builds/sphinxsearch-rel21/ubuntu/dists/trusty/Release.gpg  Unable to connect to ppa.launchpad.net:http:
W: Failed to fetch http://ppa.launchpad.net/builds/sphinxsearch-rel22/ubuntu/dists/trusty/Release.gpg  Unable to connect to ppa.launchpad.net:http:
W: Failed to fetch http://ppa.launchpad.net/webupd8team/java/ubuntu/dists/trusty/Release.gpg  Unable to connect to ppa.launchpad.net:http:
W: Some index files failed to download. They have been ignored, or old ones used instead.
## Managed by Chef on packer-5845ab1f-4d5f-541f-111b-54a89bd64361.c.eco-emissary-99515.internal :heart_eyes_cat:
## cookbook:: travis_build_environment
##     file:: templates/default/etc/cloud/templates/hosts.tmpl.erb
127.0.0.1 localhost nettuno travis vagrant
127.0.1.1 travis-job-7ea78599-c058-46aa-940f-66e16ea19f55 travis-job-7ea78599-c058-46aa-940f-66e16ea19f55 ip4-loopback trusty64
```

**Comment by BanzaiMan** [comment]
Recent apt failures are discussed elsewhere.

**Comment by mikesparr** [comment]
I think this is related to the PHP Docker image indicated: https://github.com/docker-library/php/pull/542

Another PHP image deploys fine and it has to do with attempting to install a php module using `apt` which is now explicitly blocked by PHP's image maintainers. It first looked like Travis CI network issue with apt ppas, but I think PHP's image and this pull request they implemented is the cause.

**Comment by and3rson** [comment]
Having exact same issue now.

```
Get:131 http://ppa.launchpad.net/couchdb/stable/ubuntu trusty/main Translation-en [644 B]
Fetched 36.7 MB in 6s (5,408 kB/s)
Reading package lists... Done
W: http://ppa.launchpad.net/couchdb/stable/ubuntu/dists/trusty/Release.gpg: Signature by key 15866BAFD9BCC4F3C1E0DFC7D69548E1C17EAB57 uses weak digest algorithm (SHA1)
E: Failed to fetch http://us-central1.gce.archive.ubuntu.com/ubuntu/dists/trusty-updates/main/binary-amd64/Packages.bz2  Hash Sum mismatch
E: Some index files failed to download. They have been ignored, or old ones used instead.
```

`group: deprecated-2017Q4` doesn't help.

The terrible thing is that all the crashes are random: we have jobs for PyPy, Python 2.x & several different Python 3.x. Sometimes 2.x fails, sometimes 3.x, sometimes PyPy, sometimes both.

**Comment by itkovian** [comment]
Also seeing this in a xenial build. 
~~~~
$ sudo apt-get -qq update
$ sudo apt-get install libslurm-dev
E: Could not get lock /var/lib/dpkg/lock - open (11: Resource temporarily unavailable)
E: Unable to lock the administration directory (/var/lib/dpkg/), is another process using it?
The command "sudo apt-get install libslurm-dev" failed and exited with 100 during .
~~~~

**Comment by michaeljfazio** [comment]
Can confirm Xenial distro is suffering from this.

**Comment by gsauthof** [comment]
@itkovian - what you are seeing is a different issue.

This issue is about apt-get reporting a bad signature.

Your issue is about apt-get failing to acquire a lock.

**Comment by SpaceboyRoss01** [comment]
Its appeared again:
```
Reading package lists... Done
W: GPG error: https://packagecloud.io/github/git-lfs/ubuntu trusty InRelease: The following signatures couldn't be verified because the public key is not available: NO_PUBKEY 6B05F25D762E3157
E: The repository 'https://packagecloud.io/github/git-lfs/ubuntu trusty InRelease' is not signed.
N: Updating from such a repository can't be done securely, and is therefore disabled by default.
N: See apt-secure(8) manpage for repository creation and user configuration details.
W: http://ppa.launchpad.net/couchdb/stable/ubuntu/dists/trusty/Release.gpg: Signature by key 15866BAFD9BCC4F3C1E0DFC7D69548E1C17EAB57 uses weak digest algorithm (SHA1)
The command "sudo apt update" failed and exited with 100 during .
```

**Comment by PhilippGrulich** [comment]
Its appeared again...

**Comment by cosmos-sajal** [comment]
It has appeared again...

**Comment by mxschmitt** [comment]
Same for us
